src/main/java/com/example/demo/services/impl/UserServiceImpl.java

```diff
@@ -51,20 +67,38 @@ public User getUserById(final long id) {
     }
 
     @Override
-    public void addUser(final UserCreation user) {
+    public boolean addUser(final UserCreation user) {
+        if (user.getAge() < 0) {
+            log.error(ageValidError);
+            throw new RuntimeException(ageValidError);
+        }
         userRepository.save(userEntityMapper.map(user));
+        return true;
     }
 
     @Override
-    public void updateUser(final UserUpdate userUpdate) {
+    public boolean updateUser(final UserUpdate userUpdate) {
+        if (userUpdate.getAge() < 0) {
+            log.error(ageValidError);
+            throw new RuntimeException(ageValidError);
+        }
+
         final LocalDateTime creationDate = userRepository.getById(userUpdate.getId()).getCreatedAt();
         final UserEntity updatedUserEntity = userEntityMapper.map(userUpdate);
         updatedUserEntity.setCreatedAt(creationDate);
```

Comment:
You don't need to update your time stamps manually, as your annotation will do you for you. The best approach to get user from the db, then set new values, then save it